src/rgw/rgw_sal_motr.cc

```diff
@@ -2649,6 +2664,89 @@ int MotrAtomicWriter::process(bufferlist&& data, uint64_t offset)
   return this->write();
 }
 
+int MotrObject::update_null_version_index(const DoutPrefixProvider *dpp, rgw_bucket_dir_entry& ent)
+{
+  int rc, ret_code = -ENOENT;
+  string tenant_bkt_name = get_bucket_name(this->get_bucket()->get_tenant(), this->get_bucket()->get_name());
+
+  string bucket_index_iname = "motr.rgw.bucket.index." + tenant_bkt_name;
+  std::string null_version_key = this->get_name() + "/null";
+  bufferlist::const_iterator iter;
+  bufferlist bl, bl_null_idx_val;
+  rgw_bucket_dir_entry current_null_key_ref;
+  current_null_key_ref.key.name = this->get_name();
+  current_null_key_ref.key.instance = this->get_instance();
+  current_null_key_ref.encode(bl_null_idx_val);
+  // get an entry from the cache
+  if (this->store->get_obj_meta_cache()->get(dpp, null_version_key, bl)) {
+    // Cache miss !
+    ret_code = this->store->do_idx_op_by_name(bucket_index_iname,
```

Comment:
Why get from index is required?

Reply:
As per above comment, we have removed the cache lookup. To read the index key (obj/null) we have keep the motr call here.